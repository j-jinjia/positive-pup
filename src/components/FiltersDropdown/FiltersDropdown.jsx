import "./FiltersDropdown.scss";
import filterIcon from "../../assets/svgs/filter-icon.svg";
import filterCross from "../../assets/svgs/filter-cross.svg";
import { useState } from "react";

const FiltersDropdown = ({
  handleAgeFilterSelect,
  handleDurationFilterSelect,
}) => {
  const [showFilterMenu, setShowFilterMenu] = useState(false);
  const [activeAgeInput, setActiveAgeInput] = useState(null);
  const [activeDurationInput, setActiveDurationInput] = useState(null);

  const handleClick = () => {
    setShowFilterMenu(!showFilterMenu);
  };

  const handleAgeFilterChange = (event) => {
    setActiveAgeInput(event.target.id);
  };

  const handleDurationFilterChange = (event) => {
    setActiveDurationInput(event.target.id);
  };

  const filtersDropdownMenu = (
    <div className="filters-dropdown__menu">
      <button className="filters-dropdown__cross">
        <img onClick={handleClick} src={filterCross} />
      </button>
      <h3 className="filters-dropdown__heading">Suitable Age</h3>
      <div className="filters-dropdown__filter-options">
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleAgeFilterSelect}
            onChange={handleAgeFilterChange}
            name="suitable-age"
            type="radio"
            id="all"
            checked={activeAgeInput === "all"}
          />
          <label className="filters-dropdown__label" htmlFor="all">
            All ages
          </label>
        </span>
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleAgeFilterSelect}
            onChange={handleAgeFilterChange}
            name="suitable-age"
            type="radio"
            id="infant"
            checked={activeAgeInput === "infant"}
          />
          <label className="filters-dropdown__label" htmlFor="infant">
            Infant (10 weeks &amp; above)
          </label>
        </span>
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleAgeFilterSelect}
            onChange={handleAgeFilterChange}
            name="suitable-age"
            type="radio"
            id="junior"
            checked={activeAgeInput === "junior"}
          />
          <label className="filters-dropdown__label" htmlFor="junior">
            Junior (16 weeks &amp; above)
          </label>
        </span>
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleAgeFilterSelect}
            onChange={handleAgeFilterChange}
            name="suitable-age"
            type="radio"
            id="mature"
            checked={activeAgeInput === "mature"}
          />
          <label className="filters-dropdown__label" htmlFor="mature">
            Mature (1 year &amp; above)
          </label>
        </span>
      </div>
      <h3 className="filters-dropdown__heading">Duration</h3>
      <div className="filters-dropdown__filter-options">
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleDurationFilterSelect}
            onChange={handleDurationFilterChange}
            name="duration"
            type="radio"
            id="short"
            checked={activeDurationInput === "short"}
          />
          <label className="filters-dropdown__label" htmlFor="short">
            {" "}
            &lt; 1 month
          </label>
        </span>
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleDurationFilterSelect}
            onChange={handleDurationFilterChange}
            name="duration"
            type="radio"
            id="medium"
            checked={activeDurationInput === "medium"}
          />
          <label className="filters-dropdown__label" htmlFor="medium">
            1 to 3 months
          </label>
        </span>
        <span>
          <input
            className="filters-dropdown__input"
            onClick={handleDurationFilterSelect}
            onChange={handleDurationFilterChange}
            name="duration"
            type="radio"
            id="long"
            checked={activeDurationInput === "long"}
          />
          <label className="filters-dropdown__label" htmlFor="long">
            3+ months
          </label>
        </span>
      </div>
    </div>
  );

  return (
    <div className="filters-dropdown">
      <button className="filters-dropdown__button" onClick={handleClick}>
        Filters
        <img className="filters-dropdown__button-image" src={filterIcon} />
      </button>

      {showFilterMenu && filtersDropdownMenu}
    </div>
  );
};

export default FiltersDropdown;
