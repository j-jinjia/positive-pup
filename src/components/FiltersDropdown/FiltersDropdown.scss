@use "../../styles/_variables.scss" as *;

.filters-dropdown {
  &__menu {
    border: solid 1px $colour-secondary-background;
    background-color: $colour-secondary;
    position: relative;
  }

  &__cross {
    background-color: transparent;
    border-style: none;
    padding: 1rem;
    position: absolute;
    top: 0;
    right: 0;
  }

  &__heading {
    font-size: $font-size-smallest;
    font-family: $font-general;
    color: $colour-font-general;
    padding: 1rem;
    padding-left: 2rem;
  }

  &__filter-options {
    display: flex;
    flex-direction: column;
    padding-left: 2rem;
    padding-right: 3rem;
  }

  &__label {
    font-size: $font-general;
    font-family: $font-general;
    padding-left: 1rem;
    color: $colour-font-paragraph;
  }

  &__input {
    margin: 0.5rem;
    border: solid 1px $colour-secondary-background;
    background-color: transparent;
  }

  &__input:checked {
    color: $colour-primary;
  }
}

.filters-dropdown-button {
  border: solid 1px $colour-secondary-background;
  background-color: $colour-secondary;
  padding: 1rem;
  border-radius: 0.5rem;

  &__image {
    margin-left: 0.5rem;
  }
}
