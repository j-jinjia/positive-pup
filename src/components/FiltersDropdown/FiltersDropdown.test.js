import { customRender } from "../../utils/testUtils";
import FiltersDropdown from "./FiltersDropdown";

it("should match the FiltersDropdown component snapshot", () => {
  const { container } = customRender(<FiltersDropdown />);

  expect(container).toMatchSnapshot();
});
