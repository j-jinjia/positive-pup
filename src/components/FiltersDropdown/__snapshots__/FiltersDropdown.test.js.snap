// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the FiltersDropdown component snapshot 1`] = `
<div>
  <button
    class="filter-dropdown-button"
  >
    Filters
    <img
      class="filter-dropdown-button__image"
      src="filter-icon.svg"
    />
  </button>
</div>
`;
